### ISSUE TITLE
Deinitialisation

### ISSUE THREAD

**Issue by fmontesi** [jolie/jolie#78]
Sometimes it is useful to specify code that should run when a service is shut down. Right now, we have to do this manually by implementing something like a shutdown operation that is hopefully called by the embedder. However, this does not catch, e.g., SIGTERM (ctrl-c).

We could consider having a deinit procedure, the dual of the init procedure:

```
deinit { /* your code here */ }
```

deinit would be called when the interpreter is asked to "exit", either by using the exit keyword or via callExit@RuntimeService. Use cases include:
- Telling a registry that the service is no longer going to be available.
- Closing some resources owned by the service.
- Cleaning up a Database.
- Etc.

**Comment by Ezbob** [comment]
So basically a shutdown hook for the microservice? 
I would probably called it `onShutdown { ... }` or simply `shutdown { ... }` seems to make more intuitive sense for me.

**Comment by klag** [comment]
Yes it is an important primitive. I also don't like the keyword deinit, maybe shutdown as Ezbob said or finalize.

It could be also useful to provide the RuntimeService with an interface for allowing remote control on the microservice (where the shutdown operation could be included). The developer could exhibit such an interface by aggregating the RuntimeService in one of the inputPorts.

**Comment by fmontesi** [comment]
Yeah, I also started thinking about other keywords, deinit sounds a bit unintuitive. I like `shutdown`.

@klag For what you say, how about simply having an `exit` operation in RuntimeService that acts as if we called `exit` in the embedder? We already have this functionality in `callExit@RuntimeService` but it takes the local location of the target as parameter, so I don't think it would serve your purpose.
